### DISCUSSION TITLE
Json null value

### DISCUSSION THREAD

**Discussion by agamm** [prisma/prisma#17139] [Q&A]
I have the following schema:
```
  myField              Json?                          @default("{}")
```
(notice the `?`)

And when I try to insert a null value to that json field I get:
`...<field> must not be null. Please use undefined instead`

Any way I can make it work without casting it to undefined?

**Comment by nurul3101** [comment]
Hello @agamm 👋 

You can use [Prisma.JsonNull](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#using-null-values) to add a null in the JSON column.

Here's an example of how it would look like:
```typescript
import { Prisma, PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

// A `main` function so that you can use async/await
async function main() {
  // ... you will write your Prisma Client queries here

  await prisma.user.create({
    data: {
      fields: Prisma.JsonNull,
    },
  });

  const allUsers = await prisma.user.findMany();
  console.log(allUsers);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

**Reply by agamm** [reply]
I got: ` `object` ("[object Object]") cannot be serialized as JSON. Please only return JSON serializable data types.`

**Reply by nurul3101** [reply]
Can you share the full query that you are invoking?

**Reply by agamm** [reply]
Yes of course, this is after the change of using what you suggested:
```
await prisma.website.upsert({
    update: {myField: data.myField || Prisma.JsonNull...},
    create:  {myField: data.myField || Prisma.JsonNull...},
  });
  ```
  In my case the `myField` is null.

**Reply by nurul3101** [reply]
Instead of adding an Or ( || ) condition, can you try checking the value of data.myField before, and if its value is null, pass only Prisma.JsonNull value?

**Reply by agamm** [reply]
The problem was that I needed to json encode & decode it in nextjs for it to not spit an error for serializing.

**Reply by nurul3101** [reply]
Ah, thanks for getting back and letting us know! Glad to hear that you were able to fix it! 🚀

**Reply by agamm** [reply]
I really appreciate the helpfulness!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
